web/source/keymanweb.js

```diff
@@ -1977,62 +2131,242 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
+
+          // If we just added a new input alias, some languages will mess up our touch-based layout scheme
+          // if we don't update the touch elements.
+          window.setTimeout(function() {
+            keymanweb.listInputs();
+
+            for(var k = 0; k < keymanweb.sortedInputs.length; k++) {
+              if(keymanweb.sortedInputs[k]['kmw_ip']) {
+                keymanweb.updateInput(keymanweb.sortedInputs[k]['kmw_ip']);
+              }
+            }
+          }, 1);
+        } else {
+          keymanweb.enableInputElement(Pelem);
+        }
+      }
+    }
+
+    /**
+     * Function    DisableControl
+     * Scope       Private   
+     * Parameters  Pelem   Element to be disabled
+     * Returns     None
+     * Description Legacy:  Disable KMW control element 
+     */    
+    keymanweb.DisableControl = function(Pelem)
+    {
+      if(!keymanweb.isAttached(Pelem) || !Pelem._kmwAttachment.legacy) {
+        var Lc = {LControl:Pelem, LEnabled:0, LDefaultInternalName:'-'};
+        keymanweb._Controls=keymanweb._push(keymanweb._Controls,Lc);
+        if(keymanweb.isAttached(Pelem)) {
+          Pelem._kmwAttachment.legacy = Lc;
+        }
+      } else {
+        Pelem._kmwAttachment.legacy.LEnabled = 0;
+      }
+    }
+
+    /**
+     * Function    EnableControl
+     * Scope       Private   
+     * Parameters  Pelem   Element to be enabled
+     * Returns     None
+     * Description Legacy:  Enable KMW control element 
+     */    
     keymanweb.EnableControl = function(Pelem)
     {
-      var Ln, Lc;
-      for(Ln=0; Ln<keymanweb._Controls.length; Ln++) // I1511 - array prototype extended
-        if(keymanweb._Controls[Ln].LControl == Pelem)
-        {
-          keymanweb._Controls[Ln].LEnabled = 1;
-          return;
+      if(!keymanweb.isAttached(Pelem) || !Pelem._kmwAttachment.legacy) {
+        var Lc = {LControl:Pelem, LEnabled:1, LDefaultInternalName:'-'};
+        keymanweb._Controls=keymanweb._push(keymanweb._Controls,Lc);
+        if(keymanweb.isAttached(Pelem)) {
+          Pelem._kmwAttachment.legacy = Lc;
         }
-      Lc = {LControl:Pelem, LEnabled:1, LDefaultInternalName:'-'};
-      keymanweb._Controls=keymanweb._push(keymanweb._Controls,Lc);
+      } else {
+        Pelem._kmwAttachment.legacy.LEnabled = 1;
+      }
     }
     
     /**
+     * Function     setKeyboardForControl
+     * Scope        Public   
+     * @param       {Element}    Pelem    Control element 
+     * @param       {string=}    Pkbd     Keyboard (Clears the set keyboard if set to null.)  
+     * @param       {string|null=}     Plc      Language Code
+     * Description  Set default keyboard for the control 
+     */    
+    keymanweb['setKeyboardForControl'] = keymanweb.setKeyboardForControl = function(Pelem, Pkbd, Plc) {
+      /* pass null for kbd to specify no default, or '' to specify English */
+      var index = Pkbd.indexOf("Keyboard_");
+      if(index < 0 && Pkbd != '') {
```

Comment:
I think this needs a little bit of refactoring along with previous comment and subsequent `Pkbd` nullness test.